nav2_common/cmake/nav2_package.cmake

```diff
@@ -24,7 +24,7 @@ macro(nav2_package)
   endif()
 
   if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-    add_compile_options(-Wall -Wextra -Wpedantic -Werror -fPIC)
```

Comment:
What warning message is this compensating for? Hopefully we can get this option back soon.

Reply:
It's to compensate for the deprecated function warning prevalent throughout the current code for parameters. Calls such as `get_parameter_or_set` and `set_parameter_if_not_set` are deprecated. I thought we could put this option back when we merge your lifecycle branch.